src/ui/components/App.js

```diff
@@ -111,22 +132,68 @@ class App extends React.Component {
     return (
       <>
         <div id="header">
-          <div className="logo"></div>
-          <div id="status"></div>
+          <div className="logo" />
+          <div id="status" ref={this.statusRef} />
```

Comment:
I tried to have the error state in the landing page itself as an error message above the input element. But the error message comes from `protocol.socket` :

```
function setStatus(text) {
  document.getElementById("status").innerText = text;
}
```

This function uses `innerText`, and I was not able to get the `innerText` value in `componentDidMount` or `componentDidUpdate`. I will try a bit more, else will revert all unnecessary changes like `ref`, `this.state.recordingInputError`

Reply:
hmm, feels like we should move this to react :)

Reply:
yes.................... but it was a nice exercise for me to understand to workflow..